### PR TITLE
Make setting files per-world.

### PR DESCRIPTION
**Pull request by red-001**
This separates out the server and client config files. A separate config file is now saved for every world. The client config/minetest.conf is used as a fallback if a setting isn't set in the server's config.

TODO:

- [x] Finish switching server code to use the new config file.
- [x] Extend the settings menu GUI to support the new config file.
- [x] Fix the pause menu.
- [x] misc main menu fixes.

### CODE REVIEW

**Comment by rubenwardy** [comment]
minetest.setting set and /set should write to the world's conf, does this happen?

**Comment by red-001** [comment]
@rubenwardy yes settings set by mods are also saved in the world folder.

**Comment by nerzhul** [comment]
What about spliting the minetest.conf file in repository ?

**Comment by red-001** [comment]
@nerzhul Isn't there already an issue for splitting `minetest.conf.example` into smaller files?

**Comment by rubenwardy** [comment]
Splitting up the examples file is out of scope for this PR

**Comment by nerzhul** [comment]
rebase needed

File: builtin/mainmenu/common.lua [hunk 1]
```diff
@@ -303,34 +303,38 @@ function is_server_protocol_compat_or_error(server_proto_min, server_proto_max)
...
-		end
+function get_world_setting(selected, setting)
+	local world_conf = get_world_config(selected)
+	if not world_conf then return end
+	
+	local value = world_conf:get(setting)
+	if not value then -- return default
+		value = core.settings:get(setting)
+	end
+	return value
```
Comment (rubenwardy) [inline builtin/mainmenu/common.lua]:
```lua
return value or core.settings:get(setting)
```

File: builtin/mainmenu/common.lua [hunk 2]
```diff
@@ -303,34 +303,38 @@ function is_server_protocol_compat_or_error(server_proto_min, server_proto_max)
...
+end
+
+function set_world_setting(selected, setting, value)
+	local world_conf = get_world_config(selected)
+	if not world_conf then return end
+	
+	if value then
+		world_conf:set(setting, value)
+	else
```
Comment (rubenwardy) [inline builtin/mainmenu/common.lua]: could be false, probably better to throw an error in that case

```lua
assert(value == nil, "False given to set_world_setting!")
```

**Comment by paramat** [comment]
Might need adopting.
@red-001 ?

**Comment by paramat** [comment]
Needs work but neglected, closing for now.

**Comment by Fixer-007** [comment]
Please add back "Adoption needed", someone can probably continue the work.

**Comment by paramat** [comment]
I removed that label because red-001 is active again and seems to be working on PRs.

**Comment by paramat** [comment]
See https://github.com/minetest/minetest/issues/6966#issuecomment-361021470
This PR is now abandoned.

**Comment by red-001** [comment]
Replaced by #6982

**Pull request closed without merging**